File: qt/gen_qscreen_platform.cpp
// +build ignore

#define WORKAROUND_INNER_CLASS_DEFINITION_QNativeInterface__QWindowsScreen
#include <qscreen_platform.h>
#include "gen_qscreen_platform.h"

#ifndef _Bool
#define _Bool bool
#endif

void _GUID_Delete(_GUID* self, bool isSubclass) {
	if (isSubclass) {
		delete dynamic_cast<_GUID*>( self );
	} else {
		delete self;
	}
}

void type_info_Delete(type_info* self, bool isSubclass) {
	if (isSubclass) {
		delete dynamic_cast<type_info*>( self );
	} else {
		delete self;
	}
}

class MiqtVirtualQNativeInterfaceQWindowsScreen : public virtual QNativeInterface::QWindowsScreen {
public:

	MiqtVirtualQNativeInterfaceQWindowsScreen(): QNativeInterface::QWindowsScreen() {};

private:
	virtual ~MiqtVirtualQNativeInterfaceQWindowsScreen();

public:

	// cgo.Handle value for overwritten implementation
	intptr_t handle__Handle = 0;

	// Subclass to allow providing a Go implementation
	virtual HMONITOR handle() const override {
		if (handle__Handle == 0) {
			return nullptr; // Pure virtual, there is no base we can call
		}
		

		struct HMONITOR__* callback_return_value = miqt_exec_callback_QNativeInterface__QWindowsScreen_Handle(const_cast<MiqtVirtualQNativeInterfaceQWindowsScreen*>(this), handle__Handle);

		return callback_return_value;
	}

};

QNativeInterface__QWindowsScreen* QNativeInterface__QWindowsScreen_new() {
	return new MiqtVirtualQNativeInterfaceQWindowsScreen();
}

struct HMONITOR__* QNativeInterface__QWindowsScreen_Handle(const QNativeInterface__QWindowsScreen* self) {
	HMONITOR _ret = self->handle();
	return static_cast<struct HMONITOR__*>(_ret);
}

void QNativeInterface__QWindowsScreen_override_virtual_Handle(void* self, intptr_t slot) {
	dynamic_cast<MiqtVirtualQNativeInterfaceQWindowsScreen*>( (QNativeInterface__QWindowsScreen*)(self) )->handle__Handle = slot;
}

File: qt/gen_qscreen_platform.go
package qt
import (
	"unsafe"
)

		type QNativeInterface__QWindowsScreen struct {
			h uintptr
			isSubclass bool
}
		
			// NewQNativeInterface__QWindowsScreen constructs a new QNativeInterface::QWindowsScreen object.
			func NewQNativeInterface__QWindowsScreen() *QNativeInterface__QWindowsScreen {
								
				ret := newQNativeInterface__QWindowsScreen(QNativeInterface__QWindowsScreen_new())
				ret.isSubclass = true
				return ret
			}
			
			
			func (this *QNativeInterface__QWindowsScreen) Handle() *struct HMONITOR__ {
				return (*struct HMONITOR__)(unsafe.Pointer(QNativeInterface__QWindowsScreen_Handle(this.h)))
}
			func (this *QNativeInterface__QWindowsScreen) OnHandle(slot func() *struct HMONITOR__) {
					if ! this.isSubclass {
						panic("miqt: can only override virtual methods for directly constructed types")
					}
					QNativeInterface__QWindowsScreen_override_virtual_Handle(unsafe.Pointer(this.h), intptr_t(cgo.NewHandle(slot)) )
				}
				
				//export miqt_exec_callback_QNativeInterface__QWindowsScreen_Handle
				func miqt_exec_callback_QNativeInterface__QWindowsScreen_Handle(self QNativeInterface__QWindowsScreen, cb intptr_t) *struct_HMONITOR__{
					gofunc, ok := cgo.Handle(cb).Value().(func() *struct HMONITOR__)
					if !ok {
						panic("miqt: callback of non-callback type (heap corruption?)")
					}
				
			
virtualReturn := gofunc( )

return virtualReturn

				}
			